src/fondant/pipeline.py

```diff
@@ -139,6 +141,63 @@ def from_registry(
             node_pool_name=node_pool_name,
         )
 
+    def get_component_cache_key(self) -> str:
+        """Calculate a cache key representing the unique identity of this ComponentOp.
+
+        The cache key is computed based on the component specification, image hash, arguments, and
+        other attributes of the ComponentOp. It is used to uniquely identify a specific instance
+        of the ComponentOp and is used for caching.
+
+        Returns:
+            A cache key representing the unique identity of this ComponentOp.
+        """
+
+        def sorted_dict_to_json(input_dict):
+            """Convert a dictionary to a sorted JSON string.
+
+            This function recursively converts nested dictionaries to ensure all dictionaries
+            are sorted and their values are JSON-compatible (e.g., lists, dictionaries, strings,
+            numbers, booleans, or None).
+
+            Args:
+                input_dict: The dictionary to be converted.
+
+            Returns:
+                A sorted JSON string representing the dictionary.
+            """
+            if isinstance(input_dict, dict):
+                return json.dumps(input_dict, sort_keys=True)
+
+            return input_dict
```

Comment:
I think we can now remove this whole function and replace it with the `json.dumps` call.